### PR TITLE
improved map rendering performance

### PR DESCRIPTION
**Pull request by ochanje210**

### CODE REVIEW

File: shared/actions/mapActions.js [hunk 1]
```diff
@@ -19,27 +20,26 @@ export const receiveItems = (json) => {
...
+  return async (dispatch) => {
     const queries = [];
     queries.push(createQueryObject('lat', lat));
     queries.push(createQueryObject('lng', long));
     queries.push(createQueryObject('zoom', zoomLevel));
     queries.push(createQueryObject('isThumbnail', true));
-    // todo: recover this when aws is ready: const address = `${HTTP}${SERVER_ADDR}${ENDPOINT_ITEM}${queryBuilder(queries)}`;
-    // const address = `https://goober.herokuapp.com/api/items${queryBuilder(queries)}`;
-    getAccessToken().then((accessToken) => {
+    return await getAccessToken().then(async (accessToken) => {
```
Comment (festinalente91) [inline shared/actions/mapActions.js]: It seems unnecessary to use async in the then().

Comment (ochanje210) [inline shared/actions/mapActions.js]: are you sure? could you have test on it if it still works as async function when fetching the items?

Comment (festinalente91) [inline shared/actions/mapActions.js]: let me check

File: shared/components/Card.js [hunk 2]
```diff
@@ -174,21 +175,40 @@ class Card extends Component {
...
-          <ListView
-            dataSource={new ListView.DataSource({
+    if (this.props.dataSource) {
+      if (this.props.dataSource.imageUrl) {
+        this.imageUrls = [];
+        this.imageUrls.push(this.props.dataSource.imageUrl);
+        return (
+          <View style={{flex: FLEX_LIST_WRAPPER}}>
+            <ListView
+              dataSource={new ListView.DataSource({
```
Comment (yuda1124) [inline shared/components/Card.js]: spacing is weird

File: shared/components/MainHeader.js [hunk 3]
```diff
@@ -201,13 +210,13 @@ export default class MainHeader extends Component {
                   this.props.categorizeItems('SHOW_ALL');
```
Comment (yuda1124) [inline shared/components/MainHeader.js]: why are you write show_all capitals? if you wan't to use small letter, plz standardization!

File: shared/components/Card.js [hunk 4]
```diff
@@ -174,21 +175,40 @@ class Card extends Component {
...
+              removeClippedSubviews={false}
+            />
+          </View>
+        );
+      } else if (this.props.dataSource.imageUrls) {
+        return (
+          <View style={{flex: FLEX_LIST_WRAPPER}}>
+            <ListView
+              dataSource={new ListView.DataSource({
             rowHasChanged: (r1, r2) => r1 !== r2
```
Comment (yuda1124) [inline shared/components/Card.js]: spacing

**Comment by yuda1124** [comment]
approvals

**Pull request merged**